library/keystone_service_provider.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright 2016, IBM
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+DOCUMENTATION = '''
+---
+author: Elvin Tubillara
+module: keystone_service_provider
+short_description: register sp on keystone idp
+description:
+  - This module registers a keystone service provider on the keystone
+    identity provider
+options:
+  service_provider_id
+    description:
+        A globally unique id to identify the service provider
+        example -sp.id
+    required: true
+  service_provider_url
+    description:
+        URL that is found in the service provider's metadata
+        (Which is usually found in https://keystone.sp/Shibboleth.sso/metadata)
+        example -https://keystone.sp/Shibboleth.sso/SAML2/ECP
+    required: true
+  service_provider_auth_url
+    description:
+        URL that is used to authenticate with the identity provider
+        This URL should be available once the idp registered on the sp
+        example -'http://keystone.sp/v3/OS-FEDERATION/'
+                 'identity_providers/keystone-idp/protocols/saml2/auth'
+    required: true
+'''
+
+try:
```

Comment:
No, if we use shade it can take whatever auth arguments. The auth is done with another library which I think we can dynamically pass in whatever parameters..